invenio_circulation/api.py

```diff
@@ -132,6 +132,31 @@ def is_item_available_for_checkout(item_pid):
         return search_result.hits.total == 0
 
 
+def is_item_at_desk_available_for_checkout(item_pid, patron_pid):
+    """Return True if the ITEM_AT_DESK is available for loan, False instead."""
+    # Try first to accept the availability for given patron
+    except_config_exists = current_app.config.get(
+        "CIRCULATION_STATES_LOAN_ACTIVE_EXCEPT")
+    if patron_pid and except_config_exists:
```

Comment:
Why `patron_pid` is optional? I would say that we should fail if `patron_pid` is not passed. When you checkout, we should always have a patron. Do you have any use case?
Actually it looks like that below you throw an error if no `patron_pid`.